### ISSUE TITLE
Question about VQAScore calculation method in MPCF

### ISSUE THREAD

**Issue by MaoMao78777** [Kwai-Kolors/LPO#11]
I’m trying to reproduce the VQA scores reported in the MPCF CSV file (vqa_aes_clip_score_mp.csv), but my results don’t match. I’m using the clip-flant-xxl model you provided earlier. Could you please help confirm the correct implementation? (I couldn’t find the VQA score implementation in the code you provided)

Here is my code:

` def compute_vqa_score(image, caption, model, processor):                                                                                                                                                    
      device = "cuda" if torch.cuda.is_available() else "cpu"                                                                                                                                                 
                                                                                                                                                                                                              
      question = f'Does this image show "{caption}"?'                                                                                                                                                         
                                                                                                                                                                                                              
      inputs = processor(images=image, text=question, return_tensors="pt", padding=True)                                                                                                                      
      inputs = {k: v.to(device) for k, v in inputs.items()}                                                                                                                                                   
                                                                                                                                                                                                              
      Yes_token_id = 2163                                                                                                                                                                                     
      No_token_id = 465                                                                                                                                                                                       
                                                                                                                                                                                                              
      with torch.no_grad():                                                                                                                                                                                   
          outputs = model.generate(                                                                                                                                                                           
              **inputs,                                                                                                                                                                                       
              max_new_tokens=1,                                                                                                                                                                               
              output_scores=True,                                                                                                                                                                             
              return_dict_in_generate=True,                                                                                                                                                                   
          )                                                                                                                                                                                                   
                                                                                                                                                                                                              
          first_token_logits = outputs.scores[0][0].float()                                                                                                                                                   
          probs = torch.softmax(first_token_logits, dim=-1)                                                                                                                                                   
                                                                                                                                                                                                              
          p_yes = probs[Yes_token_id].item()                                                                                                                                                                  
          p_no = probs[No_token_id].item()                                                                                                                                                                    
                                                                                                                                                                                                              
          score = p_yes / (p_yes + p_no)                                                                                                                                                                      
                                                                                                                                                                                                              
      return score     `
